packages/react/portal/src/Portal.tsx

```diff
@@ -16,11 +17,14 @@ interface PortalProps extends PrimitiveDivProps {
   /**
    * An optional container where the portaled content should be appended.
    */
-  container?: HTMLElement | null;
+  container?: Element | null;
 }
 
 const Portal = React.forwardRef<PortalElement, PortalProps>((props, forwardedRef) => {
-  const { container = globalThis?.document?.body, ...portalProps } = props;
+  const { container: containerProp, ...portalProps } = props;
+  const [mounted, setMounted] = React.useState(false);
+  useLayoutEffect(() => setMounted(true), []);
+  const container = containerProp || (mounted && globalThis?.document?.body);
```

Comment:
From an end-user point-of-view don't we introduce an extra frame as React will not render the content at first execution but after a second render cycle?

Reply:
Actually users can still pass the container to `globalThis?.document?.body` to drop that extra frame. Forget about my comment